lib/AdapterLdap.php

```diff
@@ -566,67 +568,121 @@ public function isUserInVo($user, $voShortName)
+                    $spEntityId
+                )
+            );
             return [];
         }
-
-        $facilityId = $facility->getId();
 
         $resources = $this->connector->searchForEntities(
             $this->ldapBase,
-            '(&(objectClass=perunResource)(perunFacilityDn=perunFacilityId=' . $facilityId . ','
+            '(&(objectClass=perunResource)(perunFacilityDn=perunFacilityId=' . $facility->getId() . ','
             . $this->ldapBase . '))',
             [self::CAPABILITIES, self::ASSIGNED_GROUP_ID]
         );
+        if (empty($resources)) {
+            Logger::debug(
+                self::DEBUG_PREFIX . sprintf(
+                    'getResourceCapabilities - no resources found for SP with EntityID \'%s\', returning empty list of resource capabilities.',
+                    $spEntityId
+                )
+            );
+            return [];
+        }
 
         $userGroupsIds = [];
         foreach ($userGroups as $userGroup) {
-            array_push($userGroupsIds, $userGroup->getId());
+            if ($userGroup === null || $userGroup->getId() === null) {
+                Logger::debug(
+                    self::DEBUG_PREFIX . 'getResourceCapabilities - skipping user group due to null group or null group ID.'
+                );
+                continue;
+            }
+            $userGroupsIds[] = $userGroup->getId();
         }
 
-        $resourceCapabilities = [];
+        $capabilities = [];
         foreach ($resources as $resource) {
-            if (
-                !array_key_exists(self::ASSIGNED_GROUP_ID, $resource) ||
-                !array_key_exists(self::CAPABILITIES, $resource)
-            ) {
+            if (($resource[self::ASSIGNED_GROUP_ID] ?? null) === null) {
+                Logger::debug(
+                    self::DEBUG_PREFIX . 'getResourceCapabilities - skipping resource due to null resource or null assigned group ID.'
+                );
+                continue;
+            } elseif (empty($resource[self::CAPABILITIES])) {
+                Logger::debug(
+                    self::DEBUG_PREFIX . 'getResourceCapabilities - skipping resource due to empty capabilities.'
+                );
                 continue;
             }
+
             foreach ($resource[self::ASSIGNED_GROUP_ID] as $groupId) {
                 if (in_array($groupId, $userGroupsIds, true)) {
-                    foreach ($resource[self::CAPABILITIES] as $resourceCapability) {
-                        array_push($resourceCapabilities, $resourceCapability);
-                    }
+                    $capabilities = array_merge($capabilities, $resources[self::CAPABILITIES]);
                     break;
                 }
             }
         }
 
-        return $resourceCapabilities;
+        return array_unique($capabilities);
```

Comment:
```suggestion
        return array_values(array_unique($capabilities));
```
`array_unique` keeps indexes, so there are gaps, which may cause problems later